addons/web_editor/static/lib/odoo-editor/odoo-editor.js

```diff
@@ -4132,6 +4257,118 @@ var exportVariable = (function (exports) {
             }
         }
 
+        // PASTING / DROPPING
+
+        /**
+         * Prepare clipboard data (text/html) for safe pasting into the editor.
+         *
+         * @private
+         * @param {string} clipboardData
+         * @returns {string}
+         */
+         _prepareClipboardData(clipboardData) {
+            const container = document.createElement('fake-container');
+            container.innerHTML = clipboardData;
+            for (const child of [...container.childNodes]) {
+                this._cleanForPaste(child);
+            }
+            return container.innerHTML;
+        }
+        /**
+         * Prepare clipboard data (text/plain) for safe pasting into the editor.
+         *
+         * @private
+         * @param {string} clipboardData
+         * @returns {string}
+         */
+        _prepareTextClipboardData(clipboardData) {
+            const isXML = !!clipboardData.match(/<[a-z]+[a-z0-9-]*( [^>]*)*>[\s\S\n\r]*<\/[a-z]+[a-z0-9-]*>/i);
+            const isJS = !isXML && !!clipboardData.match(/\(\);|this\.|self\.|function\s?\(|super\.|[a-z0-9]\.[a-z].*;/i);
+
+            const container = document.createElement('fake-container');
+            const pre = document.createElement('pre');
+            pre.innerHTML = clipboardData.trim()
+                .replace(/</g, '&lt;').replace(/>/g, '&gt;')
+                // Get that text as an array of text nodes separated by <br> where
+                // needed.
+                .replace(/(\n+)/g, '<br>');
+
+            if (isJS || isXML) {
+                container.appendChild(pre);
```

Comment:
I don't get what the problem is. I pasted the text and it pastes it as is. It wraps it in `<pre>` which was the spec at the time of we3 and I assumed it didn't change. For the record, the code is the same as back then and it had been validated.
I could change the code but it would change the spec.

Reply:
I don't know the spec (cf the missing commit message discussion) but I don't see why we should insert text inside a `<pre>` tag. When somebody copy/paste text, the result should not change because it contains a specific character in the message, it should appear as it was in the source text (and technically that means escaping the content).

Reply:
Look that's fine by me, this is code that used to be in Odoo (I believe it was written by @Gorash originally) and that I just restored. It does seem weird functionally to me, though not dramatically.
How does that work though because the commit you're reviewing was already reviewed and merged, this PR is just a lib update. @dmo-odoo What would be the process here?